mlx/primitives.cpp

```diff
@@ -1338,6 +1338,66 @@ std::pair<array, int> LogicalNot::vmap(
   return {logical_not(inputs[0], stream()), axes[0]};
 }
 
+std::vector<array> LogicalAnd::vjp(
+    const std::vector<array>& primals,
+    const array& cotan,
+    const std::vector<int>& argnums) {
+  assert(primals.size() == 2);
+  assert(argnums.size() == 2);
```

Comment:
This argnums should not need to have size 2. It can also be 1. Take a look at the vjp / jvp for `Add`.